### PR TITLE
Create database on save

### PR DESCRIPTION
**Pull request by mayya-sharipova**
An option createDBOnSave was added:
whether to create a new database during save operation

BugzID: 64055

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -287,6 +287,7 @@ jsonstore.rdd.minInPartition|10|the min rows in a partition.
 jsonstore.rdd.requestTimeout|100000| the request timeout in milli-second
 bulkSize|20| the bulk save size
 schemaSampleSize| -1 | the sample size for RDD schema discovery. 1 means we are using only first document for schema discovery; -1 means all documents; 0 will be treated as 1; any number N means min(N, total) docs 
+createDBOnSave|"false"| whether to create a new database during save operation. If false, a database should alreasy exist. If true, a new database will be created. If true, and a database with a provided name already exists, an error will be raised.
```
Comment (HolgerKache) [inline README.md]: s/alreasy/already

File: README.md [hunk 2]
```diff
@@ -302,6 +303,8 @@ index||cloudant search index w/o the database name. only used for load data with
 path||cloudant: as database name if database is not present
 schemaSampleSize|-1| the sample size used to discover the schema for this temp table. -1 scans all documents
 bulkSize|20| the bulk save size
+createDBOnSave|"false"| whether to create a new database during save operation. If false, a database should alreasy exist. If true, a new database will be created. If true, and a database with a provided name already exists, an error will be raised. 
+
```
Comment (HolgerKache) [inline README.md:307]: s/alreasy/already

**Comment by HolgerKache** [comment]
@mayya-sharipova I'm currently not able to test this code because of a local build error: 

```
java.lang.RuntimeException: deduplicate: different file contents found in the following:
/Users/holger/.ivy2/cache/scala_2.10/sbt_0.13/org.spark-packages/sbt-spark-package/jars/sbt-spark-package-0.2.3.jar:sbt/sbt.autoplugins
/Users/holger/.ivy2/cache/scala_2.10/sbt_0.13/com.eed3si9n/sbt-assembly/jars/sbt-assembly-0.12.0.jar:sbt/sbt.autoplugins
```

That is on `sbt assembly` in the root folder. Do you recognize this and know how to fix it?

Just looking through the code, though, I had one question. I see we use createDBOnSave as string "true" or "false" instead of a boolean true/false. The same for bulkSize "100" instead of a number 100. Does Scala only support string options or can we improve on that?

**Comment by mayya-sharipova** [comment]
@HolgerKache Thanks for looking at this. I will correct typos. 
About build, do you do "sbt assembly" in a root folder, or in the folder "cloudant-spark-sql"? We should do it in the folder "cloudant-spark-sql".

About  using strings "true" and "false", and numbers "100" - yes we have to use strings. It is because our parameters dictionary is of type: `parameters: Map[String, String]`.  https://github.com/cloudant-labs/spark-cloudant/blob/64055_create_database_on_save/cloudant-spark-sql/src/main/scala/com/cloudant/spark/common/JsonStoreConfigManager.scala#L140. The type of `Map[String, String]` for parameters is defined by spark for RelationProvider, and we can't change this: https://github.com/cloudant-labs/spark-cloudant/blob/64055_create_database_on_save/cloudant-spark-sql/src/main/scala/com/cloudant/spark/DefaultSource.scala#L100

**Comment by HolgerKache** [comment]
+1

All tested and verified. Looks good!

**Pull request merged**